internal/awsservice/cloudwatchlogs.go

```diff
@@ -6,37 +6,73 @@ package awsservice
 import (
 	"context"
 	"errors"
+	"fmt"
 	"log"
 	"strings"
-	"testing"
 	"time"
 
 	"github.com/aws/aws-sdk-go-v2/aws"
-	"github.com/aws/aws-sdk-go-v2/config"
 	"github.com/aws/aws-sdk-go-v2/service/cloudwatchlogs"
 	"github.com/aws/aws-sdk-go-v2/service/cloudwatchlogs/types"
-	"github.com/stretchr/testify/assert"
 )
 
-const allowedRetries = 5
+// catch ResourceNotFoundException when deleting the log group and log stream, as these
+// are not useful exceptions to log errors on during cleanup
+var rnf *types.ResourceNotFoundException
 
-// TODO: Refactor Structure and Interface for more easier follow that shares the same session
-var (
-	ctx context.Context
-	cwl *cloudwatchlogs.Client
-)
+type cwlAPI interface {
+	// ValidateNumberOfLogsFound takes a log group and log stream, and fetches the log events via the GetLogEvents
+	// API for all the logs since a given timestamp, and checks if the number of log events matches
+	// the expected value.
+	ValidateNumberOfLogsFound(logGroup, logStream string, numExpectedLogs int, since time.Time) error
+
+	// ValidateLogsInOrder takes a log group, log stream, a list of specific log lines and a timestamp.
+	// It should query the given log stream for log events, and then confirm that the log lines that are
+	// returned match the expected log lines. This also sanitizes the log lines from both the output and
+	// the expected lines input to ensure that they don't diverge in JSON representation (" vs ')
+	ValidateLogsInOrder(logGroup, logStream string, logLines []string, since time.Time) error
+
+	// DeleteLogGroupAndLogStream cleans up a log group and stream by name. This gracefully handles
+	// ResourceNotFoundException errors from calling the APIs
+	DeleteLogGroupAndLogStream(logGroup, logStream string)
+
+	// DeleteLogStream cleans up a log  stream by name. This gracefully handles
+	// ResourceNotFoundException errors from calling the APIs
+	DeleteLogStream(logGroup, logStream string)
+
+	// DeleteLogGroup cleans up a log group  by name. This gracefully handles
+	// ResourceNotFoundException errors from calling the APIs
+	DeleteLogGroup(logGroup string)
+
+	// IsLogGroupExist confirms whether the logGroupName exists or not
+	IsLogGroupExist(logGroup string) bool
+}
+
+type cloudwatchLogConfig struct {
```

Comment:
nit - why is this named config? This isn't a configuration struct. It's a wrapper around an SDK client

Reply:
Hmm. I guess at that time I use Config as a name for  the implementation details. However, I'm fine to changing to `{{service}}.SDK`